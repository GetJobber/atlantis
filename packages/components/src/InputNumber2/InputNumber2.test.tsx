import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { InputNumber2 } from ".";

it("renders a InputNumber2", () => {
  const { container } = render(<InputNumber2 text="Foo" />);
  expect(container).toMatchSnapshot();
});

it("renders a loud InputNumber2", () => {
  const { container } = render(<InputNumber2 text="Foo" loud={true} />);
  expect(container).toMatchSnapshot();
});

test("it should call the handler with the new value", () => {
  const clickHandler = jest.fn();
  const text = "Foo";
  const { getByText } = render(
    <InputNumber2 onClick={clickHandler} text={text} />,
  );

  fireEvent.click(getByText(text));
  expect(clickHandler).toHaveBeenCalled();

  // E.g. If you need a change event, rather than a click event:
  //
  // fireEvent.change(getByLabelText(placeholder), {
  //   target: { value: newValue },
  // });
  // expect(changeHandler).toHaveBeenCalledWith(newValue);
});
