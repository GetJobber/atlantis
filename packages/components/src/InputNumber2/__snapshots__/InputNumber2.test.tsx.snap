// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a InputNumber2 1`] = `
<div>
  <div
    class="inputNumber2"
  >
    Foo
  </div>
</div>
`;

exports[`renders a loud InputNumber2 1`] = `
<div>
  <div
    class="inputNumber2 bold"
  >
    Foo
  </div>
</div>
`;
